"use strict";(self.webpackChunk=self.webpackChunk||[]).push([[64544],{603905:(e,r,n)=>{n.d(r,{Zo:()=>p,kt:()=>m});var a=n(667294);function o(e,r,n){return r in e?Object.defineProperty(e,r,{value:n,enumerable:!0,configurable:!0,writable:!0}):e[r]=n,e}function t(e,r){var n=Object.keys(e);if(Object.getOwnPropertySymbols){var a=Object.getOwnPropertySymbols(e);r&&(a=a.filter((function(r){return Object.getOwnPropertyDescriptor(e,r).enumerable}))),n.push.apply(n,a)}return n}function l(e){for(var r=1;r<arguments.length;r++){var n=null!=arguments[r]?arguments[r]:{};r%2?t(Object(n),!0).forEach((function(r){o(e,r,n[r])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(n)):t(Object(n)).forEach((function(r){Object.defineProperty(e,r,Object.getOwnPropertyDescriptor(n,r))}))}return e}function s(e,r){if(null==e)return{};var n,a,o=function(e,r){if(null==e)return{};var n,a,o={},t=Object.keys(e);for(a=0;a<t.length;a++)n=t[a],r.indexOf(n)>=0||(o[n]=e[n]);return o}(e,r);if(Object.getOwnPropertySymbols){var t=Object.getOwnPropertySymbols(e);for(a=0;a<t.length;a++)n=t[a],r.indexOf(n)>=0||Object.prototype.propertyIsEnumerable.call(e,n)&&(o[n]=e[n])}return o}var i=a.createContext({}),d=function(e){var r=a.useContext(i),n=r;return e&&(n="function"==typeof e?e(r):l(l({},r),e)),n},p=function(e){var r=d(e.components);return a.createElement(i.Provider,{value:r},e.children)},c={inlineCode:"code",wrapper:function(e){var r=e.children;return a.createElement(a.Fragment,{},r)}},u=a.forwardRef((function(e,r){var n=e.components,o=e.mdxType,t=e.originalType,i=e.parentName,p=s(e,["components","mdxType","originalType","parentName"]),u=d(n),m=o,g=u["".concat(i,".").concat(m)]||u[m]||c[m]||t;return n?a.createElement(g,l(l({ref:r},p),{},{components:n})):a.createElement(g,l({ref:r},p))}));function m(e,r){var n=arguments,o=r&&r.mdxType;if("string"==typeof e||o){var t=n.length,l=new Array(t);l[0]=u;var s={};for(var i in r)hasOwnProperty.call(r,i)&&(s[i]=r[i]);s.originalType=e,s.mdxType="string"==typeof e?e:o,l[1]=s;for(var d=2;d<t;d++)l[d]=n[d];return a.createElement.apply(null,l)}return a.createElement.apply(null,n)}u.displayName="MDXCreateElement"},918907:(e,r,n)=>{n.r(r),n.d(r,{assets:()=>d,contentTitle:()=>s,default:()=>u,frontMatter:()=>l,metadata:()=>i,toc:()=>p});n(667294);var a=n(603905);function o(){return o=Object.assign||function(e){for(var r=1;r<arguments.length;r++){var n=arguments[r];for(var a in n)Object.prototype.hasOwnProperty.call(n,a)&&(e[a]=n[a])}return e},o.apply(this,arguments)}function t(e,r){if(null==e)return{};var n,a,o=function(e,r){if(null==e)return{};var n,a,o={},t=Object.keys(e);for(a=0;a<t.length;a++)n=t[a],r.indexOf(n)>=0||(o[n]=e[n]);return o}(e,r);if(Object.getOwnPropertySymbols){var t=Object.getOwnPropertySymbols(e);for(a=0;a<t.length;a++)n=t[a],r.indexOf(n)>=0||Object.prototype.propertyIsEnumerable.call(e,n)&&(o[n]=e[n])}return o}const l={id:"error-handling",title:"Gesti\xf3n de errores"},s=void 0,i={unversionedId:"Concepts/error-handling",id:"version-20-R4/Concepts/error-handling",title:"Gesti\xf3n de errores",description:"El manejo de errores es el proceso de anticipar y responder a los errores que puedan ocurrir en su aplicaci\xf3n. 4D soporta de forma completa la detecci\xf3n y notificaci\xf3n de errores en tiempo de ejecuci\xf3n, as\xed como el an\xe1lisis de sus condiciones.",source:"@site/i18n/es/docusaurus-plugin-content-docs/version-20-R4/Concepts/error-handling.md",sourceDirName:"Concepts",slug:"/Concepts/error-handling",permalink:"/docs/es/Concepts/error-handling",draft:!1,editUrl:"https://github.com/4d/docs/issues/new?title=Comment%20on%20Concepts%2Ferror-handling.md%20(20-R4)&body=Please%20enter%20your%20comment%3A",tags:[],version:"20-R4",frontMatter:{id:"error-handling",title:"Gesti\xf3n de errores"},sidebar:"docs",previous:{title:"Flujo de control",permalink:"/docs/es/Concepts/control-flow"},next:{title:"Modos interpretado y compilado",permalink:"/docs/es/Concepts/interpreted-compiled"}},d={},p=[{value:"Error o estado",id:"error-o-estado",level:2},{value:"Instalaci\xf3n de un m\xe9todo de gesti\xf3n de errores",id:"instalaci\xf3n-de-un-m\xe9todo-de-gesti\xf3n-de-errores",level:2},{value:"Alcance y componentes",id:"alcance-y-componentes",level:3},{value:"Manejo de errores e el m\xe9todo",id:"manejo-de-errores-e-el-m\xe9todo",level:3},{value:"Ejemplo",id:"ejemplo",level:4},{value:"Utilizar un m\xe9todo de gesti\xf3n de errores vac\xedo",id:"utilizar-un-m\xe9todo-de-gesti\xf3n-de-errores-vac\xedo",level:3},{value:"Try(expression)",id:"tryexpression",level:2},{value:"Ejemplos",id:"ejemplos",level:3}],c={toc:p};function u(e){var{components:r}=e,l=t(e,["components"]);return(0,a.kt)("wrapper",o({},c,l,{components:r,mdxType:"MDXLayout"}),(0,a.kt)("p",null,"El manejo de errores es el proceso de anticipar y responder a los errores que puedan ocurrir en su aplicaci\xf3n. 4D soporta de forma completa la detecci\xf3n y notificaci\xf3n de errores en tiempo de ejecuci\xf3n, as\xed como el an\xe1lisis de sus condiciones."),(0,a.kt)("p",null,"La gesti\xf3n de errores responde a dos necesidades principales:"),(0,a.kt)("ul",null,(0,a.kt)("li",{parentName:"ul"},"descubrir y corregir posibles errores y fallos en el c\xf3digo durante la fase de desarrollo,"),(0,a.kt)("li",{parentName:"ul"},"detectar y recuperar errores inesperados en las aplicaciones desplegadas; en particular, puede sustituir los di\xe1logos de error del sistema (disco lleno, archivo perdido, etc.) por su propia interfaz.")),(0,a.kt)("p",null,"B\xe1sicamente, hay dos maneras de manejar los errores en 4D. Puede ",(0,a.kt)("a",o({parentName:"p"},{href:"#installing-an-error-handling-method"}),"instalar un m\xe9todo de gesti\xf3n de errores"),", o escribir una palabra clave ",(0,a.kt)("a",o({parentName:"p"},{href:"#tryexpression"}),(0,a.kt)("inlineCode",{parentName:"a"},"Try()"))," antes de los fragmentos de c\xf3digo que llamen a una funci\xf3n, m\xe9todo o expresi\xf3n que pueda lanzar un error."),(0,a.kt)("admonition",o({},{title:"Buenas pr\xe1cticas",type:"tip"}),(0,a.kt)("p",{parentName:"admonition"},"Es muy recomendable instalar un m\xe9todo global de gesti\xf3n de errores en 4D Server, para todo el c\xf3digo que se ejecute en el servidor. Cuando 4D Server no se ejecuta ",(0,a.kt)("a",o({parentName:"p"},{href:"/docs/es/Admin/cli"}),"headless")," (es decir, se lanza con su ",(0,a.kt)("a",o({parentName:"p"},{href:"/docs/es/ServerWindow/overview"}),"ventana de administraci\xf3n"),"), este m\xe9todo evitar\xeda que se mostraran cajas de di\xe1logo inesperadas en la m\xe1quina servidor. En modo sin interfaz, los errores se registran en el archivo ",(0,a.kt)("a",o({parentName:"p"},{href:"/docs/es/Debugging/debugLogFiles#4ddebuglogtxt-standard"}),"4DDebugLog")," para su posterior an\xe1lisis.")),(0,a.kt)("h2",o({},{id:"error-o-estado"}),"Error o estado"),(0,a.kt)("p",null,"Muchas funciones de clase 4D, como ",(0,a.kt)("inlineCode",{parentName:"p"},"entity.save()")," o ",(0,a.kt)("inlineCode",{parentName:"p"},"transporter.send()"),", devuelven un objeto ",(0,a.kt)("em",{parentName:"p"},"status"),'. Este objeto se utiliza para almacenar errores "predecibles" en el contexto de ejecuci\xf3n, por ejemplo, una contrase\xf1a no v\xe1lida, una entidad bloqueada, etc., que no detienen la ejecuci\xf3n del programa. Esta categor\xeda de errores puede ser manejada por el c\xf3digo habitual.'),(0,a.kt)("p",null,'Otros errores "imprevisibles" son el error de escritura en el disco, el fallo de la red o, en general, cualquier interrupci\xf3n inesperada. Esta categor\xeda de errores genera excepciones y debe gestionarse mediante un m\xe9todo de gesti\xf3n de errores o una palabra clave ',(0,a.kt)("inlineCode",{parentName:"p"},"Try()"),"."),(0,a.kt)("h2",o({},{id:"instalaci\xf3n-de-un-m\xe9todo-de-gesti\xf3n-de-errores"}),"Instalaci\xf3n de un m\xe9todo de gesti\xf3n de errores"),(0,a.kt)("p",null,"En 4D, todos los errores pueden ser detectados y manejados por m\xe9todos proyecto espec\xedficos, llamados ",(0,a.kt)("strong",{parentName:"p"},"m\xe9todos de gesti\xf3n de errores")," (o ",(0,a.kt)("strong",{parentName:"p"},"m\xe9todos de intercepci\xf3n de errores"),")."),(0,a.kt)("p",null,"Una vez instalados, los manejadores de errores son llamados autom\xe1ticamente en modo interpretado o compilado en caso de error en la aplicaci\xf3n 4D o en uno de sus componentes. Se puede llamar a un manejador de errores diferente en funci\xf3n del contexto de ejecuci\xf3n (ver abajo)."),(0,a.kt)("p",null,"Para ",(0,a.kt)("em",{parentName:"p"},"instalar")," un m\xe9todo proyecto de gesti\xf3n de errores, basta con llamar al comando ",(0,a.kt)("a",o({parentName:"p"},{href:"https://doc.4d.com/4dv19/help/command/en/page155.html"}),(0,a.kt)("inlineCode",{parentName:"a"},"ON ERR CALL"))," con el nombre del m\xe9todo proyecto y (opcionalmente) el \xe1lcance como par\xe1metros. Por ejemplo:"),(0,a.kt)("pre",null,(0,a.kt)("code",o({parentName:"pre"},{className:"language-4d"}),'ON ERR CALL("IO_Errors";ek local) //Instala un m\xe9todo local de gesti\xf3n de errores\n')),(0,a.kt)("p",null,"Para dejar de interceptar los errores en un contexto de ejecuci\xf3n y devolver la mano, llame ",(0,a.kt)("inlineCode",{parentName:"p"},"ON ERR CALL")," con una cadena vac\xeda:"),(0,a.kt)("pre",null,(0,a.kt)("code",o({parentName:"pre"},{className:"language-4d"}),'ON ERR CALL("";ek local) //devuelve el control al proceso local\n')),(0,a.kt)("p",null,"El comando ",(0,a.kt)("a",o({parentName:"p"},{href:"https://doc.4d.com/4dv19/help/command/en/page704.html"}),(0,a.kt)("inlineCode",{parentName:"a"},"Method called on error"))," permite conocer el nombre del m\xe9todo instalado por ",(0,a.kt)("inlineCode",{parentName:"p"},"ON ERR CALL")," para el proceso actual. Es particularmente \xfatil en el contexto de c\xf3digo gen\xe9rico porque permite cambiar temporalmente y luego restaurar el m\xe9todo de captura de error:"),(0,a.kt)("pre",null,(0,a.kt)("code",o({parentName:"pre"},{className:"language-4d"}),' $methCurrent:=Method called on error(ek local)\n ON ERR CALL("NewMethod";ek local)\n  //Si no se puede abrir el documento, se genera un error\n $ref:=Open document("MyDocument")\n  //Reinstalaci\xf3n del m\xe9todo anterior\n ON ERR CALL($methCurrent;ek local)\n\n')),(0,a.kt)("h3",o({},{id:"alcance-y-componentes"}),"Alcance y componentes"),(0,a.kt)("p",null,"Se puede definir un m\xe9todo de gesti\xf3n de errores para diferentes contextos de ejecuci\xf3n:"),(0,a.kt)("ul",null,(0,a.kt)("li",{parentName:"ul"},"para el ",(0,a.kt)("strong",{parentName:"li"},"proceso actual"),"- s\xf3lo se llamar\xe1 a un gestor de errores local para los errores ocurridos en el proceso actual del proyecto actual,"),(0,a.kt)("li",{parentName:"ul"},"para ",(0,a.kt)("strong",{parentName:"li"},"toda la aplicaci\xf3n"),"- se llamar\xe1 a un gestor de errores global para todos los errores que se produzcan en el contexto de ejecuci\xf3n de la aplicaci\xf3n del proyecto actual,"),(0,a.kt)("li",{parentName:"ul"},"desde los ",(0,a.kt)("strong",{parentName:"li"},"componentes"),"- este manejador de errores se define en un proyecto local y ser\xe1 llamado para todos los errores que ocurran en los componentes cuando no hayan sido ya interceptados por un manejador de componentes.")),(0,a.kt)("p",null,"Ejemplos:"),(0,a.kt)("pre",null,(0,a.kt)("code",o({parentName:"pre"},{className:"language-4d"}),'ON ERR CALL("IO_Errors";ek local) //Instala un m\xe9todo de gesti\xf3n de errores local \nON ERR CALL("globalHandler";ek global) //Instala un m\xe9todo de gesti\xf3n de errores global \nON ERR CALL("componentHandler";ek errors from components) //Instala un m\xe9todo de gesti\xf3n de errores para los componentes\n')),(0,a.kt)("p",null,'Puede instalar un gestor de errores global que servir\xe1 como "fallback" y gestores de errores locales espec\xedficos para determinados procesos. Un gestor de errores global tambi\xe9n es \xfatil en el servidor para evitar di\xe1logos de error en el servidor cuando se ejecuta con interfaz.'),(0,a.kt)("p",null,"Se puede definir un \xfanico m\xe9todo de captura de errores para toda la aplicaci\xf3n o diferentes m\xe9todos por m\xf3dulo de aplicaci\xf3n. Sin embargo, s\xf3lo se puede instalar un m\xe9todo por contexto de ejecuci\xf3n y por proyecto."),(0,a.kt)("p",null,"Cuando se produce un error, s\xf3lo se llama a un m\xe9todo, como se describe en el siguiente diagrama:"),(0,a.kt)("p",null,(0,a.kt)("img",{alt:"gesti\xf3n de errores",src:n(119486).Z,width:"1195",height:"578"})),(0,a.kt)("h3",o({},{id:"manejo-de-errores-e-el-m\xe9todo"}),"Manejo de errores e el m\xe9todo"),(0,a.kt)("p",null,"Dentro de un m\xe9todo de gesti\xf3n de errores personalizado, tiene acceso a varios datos que le ayudar\xe1n a identificar el error:"),(0,a.kt)("ul",null,(0,a.kt)("li",{parentName:"ul"},(0,a.kt)("p",{parentName:"li"},"las variables sistema dedicadas:"),(0,a.kt)("ul",{parentName:"li"},(0,a.kt)("li",{parentName:"ul"},(0,a.kt)("inlineCode",{parentName:"li"},"Error")," (entero largo): c\xf3digo de error"),(0,a.kt)("li",{parentName:"ul"},(0,a.kt)("inlineCode",{parentName:"li"},"Error method"),"(texto): nombre del m\xe9todo que ha provocado el error"),(0,a.kt)("li",{parentName:"ul"},(0,a.kt)("inlineCode",{parentName:"li"},"Error line")," (entero largo): n\xfamero de l\xednea del m\xe9todo que ha provocado el error"),(0,a.kt)("li",{parentName:"ul"},(0,a.kt)("inlineCode",{parentName:"li"},"Error formula")," (texto): f\xf3rmula del c\xf3digo 4D (texto bruto) que est\xe1 en el origen del error.")))),(0,a.kt)("admonition",o({},{type:"info"}),(0,a.kt)("p",{parentName:"admonition"},"4D mantiene autom\xe1ticamente una serie de variables denominadas ",(0,a.kt)("strong",{parentName:"p"},"variables sistema"),", que responden a diferentes necesidades. Consulte el ",(0,a.kt)("em",{parentName:"p"},"manual de referencia del lenguaje 4D"),".")),(0,a.kt)("ul",null,(0,a.kt)("li",{parentName:"ul"},"el comando ",(0,a.kt)("a",o({parentName:"li"},{href:"https://doc.4d.com/4dv19/help/command/en/page1799.html"}),(0,a.kt)("inlineCode",{parentName:"a"},"Last errors"))," que devuelve una colecci\xf3n de la pila actual de errores ocurridos en la aplicaci\xf3n 4D. Tambi\xe9n puede utilizar el comando ",(0,a.kt)("a",o({parentName:"li"},{href:"https://doc.4d.com/4dv19/help/command/en/page1015.html"}),(0,a.kt)("inlineCode",{parentName:"a"},"GET LAST ERROR STACK"))," que devuelve la misma informaci\xf3n que los arrays."),(0,a.kt)("li",{parentName:"ul"},"el comando ",(0,a.kt)("inlineCode",{parentName:"li"},"Get call chain")," que devuelve una colecci\xf3n de objetos que describen cada paso de la cadena de llamadas a m\xe9todos dentro del proceso actual.")),(0,a.kt)("h4",o({},{id:"ejemplo"}),"Ejemplo"),(0,a.kt)("p",null,"He aqu\xed un sistema de gesti\xf3n de errores sencillo:"),(0,a.kt)("pre",null,(0,a.kt)("code",o({parentName:"pre"},{className:"language-4d"}),'//instalar el m\xe9todo de gesti\xf3n de errores\n ON ERR CALL("errorMethod")\n //... ejecutar el c\xf3digo\n ON ERR CALL("") //giving control back to 4D\n')),(0,a.kt)("pre",null,(0,a.kt)("code",o({parentName:"pre"},{className:"language-4d"}),'// m\xe9todo proyecto errorMethod\n If(Error#1006) //esto no es una interrupci\xf3n del usuario\n    ALERT("Se ha producido el error "+String(Error)+". El c\xf3digo en cuesti\xf3n es: \\""+Error formula+"\\"")\n End if\n')),(0,a.kt)("h3",o({},{id:"utilizar-un-m\xe9todo-de-gesti\xf3n-de-errores-vac\xedo"}),"Utilizar un m\xe9todo de gesti\xf3n de errores vac\xedo"),(0,a.kt)("p",null,"Si desea principalmente que la caja de di\xe1logo de error est\xe1ndar est\xe9 oculta, puede instalar un m\xe9todo de gesti\xf3n de errores vac\xedo. La variable sistema ",(0,a.kt)("inlineCode",{parentName:"p"},"Error")," puede ser probada en cualquier m\xe9todo, es decir, fuera del m\xe9todo de gesti\xf3n de errores:"),(0,a.kt)("pre",null,(0,a.kt)("code",o({parentName:"pre"},{className:"language-4d"}),'ON ERR CALL("emptyMethod") //emptyMethod exists but is empty\n$doc:=Open document( "myFile.txt")\nIf (Error=-43)\n    ALERT("File not found.")\nEnd if\nON ERR CALL("")\nEnd if\nON ERR CALL("")\n')),(0,a.kt)("h2",o({},{id:"tryexpression"}),"Try(expression)"),(0,a.kt)("p",null,"La sentencia ",(0,a.kt)("inlineCode",{parentName:"p"},"Try(expression)")," permite probar una expresi\xf3n de una sola l\xednea en su contexto de ejecuci\xf3n real (incluyendo, en particular, los valores de las variables locales) e interceptar los errores que arroje para que no se muestre el di\xe1logo de error 4D. El uso de ",(0,a.kt)("inlineCode",{parentName:"p"},"Try(expression)")," ofrece una manera f\xe1cil de manejar casos de error simples con un n\xfamero muy bajo de l\xedneas de c\xf3digo, y sin requerir un m\xe9todo de gesti\xf3n de errores."),(0,a.kt)("p",null,"La sintaxis formal de la declaraci\xf3n ",(0,a.kt)("inlineCode",{parentName:"p"},"Try(expresi\xf3n)")," es:"),(0,a.kt)("pre",null,(0,a.kt)("code",o({parentName:"pre"},{className:"language-4d"}),"\nTry (expression) : any | Undefined\n\n")),(0,a.kt)("p",null,(0,a.kt)("em",{parentName:"p"},"expresion")," puede ser toda expresi\xf3n v\xe1lida."),(0,a.kt)("p",null,"Si se produce un error durante su ejecuci\xf3n, se intercepta y no se muestra ning\xfan di\xe1logo de error, si un ",(0,a.kt)("a",o({parentName:"p"},{href:"#installing-an-error-handling-method"}),"m\xe9todo de gesti\xf3n de errores")," fue instalado o no antes de la llamada a ",(0,a.kt)("inlineCode",{parentName:"p"},"Try()"),". Si ",(0,a.kt)("em",{parentName:"p"},"expression")," devuelve un valor, ",(0,a.kt)("inlineCode",{parentName:"p"},"Try()")," devuelve el \xfaltimo valor evaluado, en caso contrario devuelve ",(0,a.kt)("inlineCode",{parentName:"p"},"Define"),"."),(0,a.kt)("p",null,"Puede gestionar el error o errores utilizando el comando ",(0,a.kt)("a",o({parentName:"p"},{href:"https://doc.4d.com/4dv20/help/command/en/page1799.html"}),(0,a.kt)("inlineCode",{parentName:"a"},"Last errors")),". Si ",(0,a.kt)("em",{parentName:"p"},"expression")," arroja un error dentro de una pila de llamadas ",(0,a.kt)("inlineCode",{parentName:"p"},"Try()"),", el flujo de ejecuci\xf3n se detiene y devuelve a la \xfaltima ejecuci\xf3n ",(0,a.kt)("inlineCode",{parentName:"p"},"Try()")," (la primera encontrada de nuevo en la pila de llamadas)."),(0,a.kt)("admonition",o({},{type:"note"}),(0,a.kt)("p",{parentName:"admonition"},"Si un ",(0,a.kt)("a",o({parentName:"p"},{href:"#installing-an-error-handling-method"}),"m\xe9todo de gesti\xf3n de errores")," es instalado por ",(0,a.kt)("em",{parentName:"p"},"expression"),", es llamado en caso de error.")),(0,a.kt)("h3",o({},{id:"ejemplos"}),"Ejemplos"),(0,a.kt)("ol",null,(0,a.kt)("li",{parentName:"ol"},"Quiere mostrar el contenido de un archivo si el archivo se puede abrir sin errores y si su contenido se puede leer. Puede escribir:")),(0,a.kt)("pre",null,(0,a.kt)("code",o({parentName:"pre"},{className:"language-4d"}),'var $text : Text\nvar $file : 4D.File := File("/RESOURCES/myFile.txt")\nvar $fileHandle : 4D.FileHandle := Try($file.open())\nIf ($fileHandle # Null)\n  $text:=Try($fileHandle.readText()) || "Error al leer el archivo"\nEnd if\n')),(0,a.kt)("ol",o({},{start:2}),(0,a.kt)("li",{parentName:"ol"},"Quiere manejar el error de dividir por cero. En este caso, se desea devolver 0 y lanzar un error:")),(0,a.kt)("pre",null,(0,a.kt)("code",o({parentName:"pre"},{className:"language-4d"}),'function divide( $p1: real; $p2: real)-> $result: real\n  if ($p2=0)\n     $result:=0 //s\xf3lo por claridad (0 es el valor por defecto para reales)\n     throw(-12345; "Divisi\xf3n por cero")\n  else\n    $result:=$p1/$p2\n  end if\n\nfunction test()\n  $result:=Try(divide($p1;$p2))\n  If (Last errors # null)\n    ALERT("Error")\n  End if\n\n')),(0,a.kt)("ol",o({},{start:3}),(0,a.kt)("li",{parentName:"ol"},"Desea gestionar tanto los errores ",(0,a.kt)("a",o({parentName:"li"},{href:"#error-or-status"}),"previsibles como los no previsibles>"),":")),(0,a.kt)("pre",null,(0,a.kt)("code",o({parentName:"pre"},{className:"language-4d"}),'var $e:=ds.Employee.new()\n$e.name:="Smith"\n$status:=Try($e.save()) //Capturar errores predecibles y no predecibles\nIf ($status.success)\n   ALERT( "Success")\nElse\n   ALERT( "Error: "+JSON Stringify($status.errors))\nEnd if\n\n')))}u.isMDXComponent=!0},119486:(e,r,n)=>{n.d(r,{Z:()=>a});const a=n.p+"assets/images/error-schema-973d40c4618e79de87f884622e4e930f.png"}}]);